tests/functional/test_data_completeness.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from tests import unittest
+
+from botocore.session import Session
+from botocore.loaders import Loader
+from botocore.exceptions import DataNotFoundError
+
+
+class LoaderWrapper(Loader):
+
+    def load_data(self, name):
+        try:
+            return super(LoaderWrapper, self).load_data(name)
+        except DataNotFoundError:
+            return None  # So we map the exception to None
+
+
+class TestDataTypes(unittest.TestCase):
+
+    def _test_data_type_is_not_lost_in_new_version(self, type_name):
+        # Make sure if a paginator and/or waiter exists in previous version,
+        # there will be a successor existing in latest version.
+        loader = LoaderWrapper()
+        for service_name in Session().get_available_services():
```

Comment:
Use test generators (search in the tests/ directory for `test_*` methods that `yield`).

Right now, this will stop as soon as a single service fails this check and we won't know if other services are susceptible to this issue.